pkg/defaulting/images.go

```diff
@@ -0,0 +1,37 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2012 Datadog, Inc.
+
+package defaulting
```

Comment:
I wonder if `defaulting` is too generic and has the risk of becoming another "utils" package where anything can be placed. Given that this is just about images, maybe it makes sense to define an `images` package instead?

Reply:
In a near futur, the idea is to have a package `defaulting` to move `datadogagent_default.go` logic from `/api/v1alpha1` to this new package.
It will be useful, if we create a dedicate repo for the CRD definitions.

Reply:
:+1: